src/shmem_team.h

```diff
@@ -0,0 +1,94 @@
+/* -*- C -*-
+ *
+ * Copyright (c) 2019 Intel Corporation. All rights reserved.
+ * This software is available to you under the BSD license.
+ *
+ * This file is part of the Sandia OpenSHMEM software package. For license
+ * information, see the LICENSE file in the top level directory of the
+ * distribution.
+ *
+ */
+
+#ifndef SHMEM_TEAM_H
+#define SHMEM_TEAM_H
+
+#include "shmemx.h"
+#include "transport.h"
+#include "uthash.h"
+
+#define N_PSYNCS_PER_TEAM   2
+#define N_PSYNC_BYTES       8
+#define PSYNC_CHUNK_SIZE    N_PSYNCS_PER_TEAM * SHMEM_SYNC_SIZE
+
+struct shmem_internal_team_t {
+    int                            my_pe;
+    int                            start, stride, size;
+    int                            psync_idx;
+    int                            psync_avail[N_PSYNCS_PER_TEAM];
+    shmemx_team_config_t           config;
+    long                           config_mask;
+    size_t                         contexts_len;
+    struct shmem_transport_ctx_t **contexts;
+};
+typedef struct shmem_internal_team_t shmem_internal_team_t;
+
+extern shmem_internal_team_t shmem_internal_team_world;
+extern shmem_internal_team_t shmem_internal_team_shared;
+
+extern long *shmem_internal_psync_pool;
+extern long *shmem_internal_psync_barrier_pool;
+
+enum shmem_internal_team_op_t {
+    SYNC = 0,
+    BCAST,
+    REDUCE,
+    COLLECT,
+    ALLTOALL
+};
+typedef enum shmem_internal_team_op_t shmem_internal_team_op_t;
+
+extern shmem_internal_team_op_t shmem_internal_team_sync_type;
+extern shmem_internal_team_op_t shmem_internal_team_bcast_type;
+extern shmem_internal_team_op_t shmem_internal_team_reduce_type;
+extern shmem_internal_team_op_t shmem_internal_team_collect_type;
+extern shmem_internal_team_op_t shmem_internal_team_alltoall_type;
```

Comment:
This looks redundant.  Why not just use the enumeration constants directly?  If these are needed (I can't see why), they should be `const` qualified.

Reply:
Hmm.. they're not needed.  Not sure what I was thinking there.